src/message/mod.rs

```diff
@@ -249,7 +99,8 @@ impl MessageManager for ChannelMessageManager {
     */
 
     fn is_block(&self) -> bool {
-        self.sender.len() >= (self.sender.capacity().unwrap() as f64 * 0.9) as usize
+        self.sender.is_full()
```

Comment:
This seems changing the behavior. But looks still fine.

Reply:
Yes. we do not use the channel to cache incoming messages: they are now cached in the messenger and when the buffer in messenger is full we simply stop receiving and this leading to the channel being blocked.

Now we can just set a small capability (I just set to 16) for the channel.